submissions/src/test/kotlin/SubmissionControllerTest.kt

```diff
@@ -0,0 +1,290 @@
+
+    private lateinit var blobContainerClient: BlobContainerClient
+    private lateinit var blobClient: BlobClient
+    private lateinit var queueClient: QueueClient
+    private lateinit var sendMessageResult: SendMessageResult
+
+    @BeforeEach
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        objectMapper = jacksonObjectMapper().registerModule(JavaTimeModule())
+
+        // Mock BlobContainerClient and BlobClient
+        blobContainerClient = mock(BlobContainerClient::class.java)
+        blobClient = mock(BlobClient::class.java)
+        `when`(blobServiceClient.getBlobContainerClient(anyString())).thenReturn(blobContainerClient)
+        `when`(blobContainerClient.getBlobClient(anyString())).thenReturn(blobClient)
+
+        // Mock QueueClient
+        queueClient = mock(QueueClient::class.java)
+        `when`(queueServiceClient.getQueueClient(anyString())).thenReturn(queueClient)
+
+        // Mock SendMessageResult
+        sendMessageResult = mock(SendMessageResult::class.java)
+
+        // Ensure blobUrl is mocked properly
+        `when`(blobClient.blobUrl).thenReturn("https://example.com/blobUrl")
+
+        // Mock the upload method
+        doNothing().`when`(blobClient).upload(any(ByteArrayInputStream::class.java), anyLong())
+
+        // Mock the queue sendMessage method
+        `when`(queueClient.sendMessage(anyString())).thenReturn(sendMessageResult)
+
+        // Mock the table createEntity method
+        doNothing().`when`(tableClient).createEntity(any())
+
+        // Ensure telemetryClient methods are mocked
+        doNothing().`when`(telemetryClient).trackEvent(anyString(), anyMap(), isNull())
+        doNothing().`when`(telemetryClient).flush()
+    }
+
+    @AfterEach
+    fun tearDown() {
+        // Reset all mocks after each test to ensure isolation
+        reset(
+            blobServiceClient,
+            queueServiceClient,
+            tableClient,
+            telemetryClient,
+            blobContainerClient,
+            blobClient,
+            queueClient
+        )
+    }
+
+
+    @Test
+    fun `submitReport should return CREATED status with content-type hl7-v2`() {
+        val data = mapOf("key" to "value")
+        val requestBody = objectMapper.writeValueAsString(data)
+
+        `when`(queueClient.sendMessage(anyString())).thenReturn(sendMessageResult)
+
+        mockMvc.perform(
+            MockMvcRequestBuilders.post("/api/v1/reports")
+                .content(requestBody)
+                .contentType(MediaType.valueOf("application/hl7-v2"))
+                .header("client_id", "testClient")
+                .header("payloadname", "testPayload")
+                .header("x-azure-clientip", "127.0.0.1")
+        )
+            .andExpect(MockMvcResultMatchers.status().isCreated)
+
+        verify(blobClient).upload(any(ByteArrayInputStream::class.java), anyLong())
+        verify(queueClient).sendMessage(anyString())
```

Comment:
Nit: could we harden these tests?  This doesn't verify that the queue message is actually correct

Reply:
added a listener and some verifys on on the inputs